avm/ptn/app/container-job-toolkit/main.bicep

```diff
@@ -0,0 +1,341 @@
+@metadata({ example: 'application/frontend:latest' })
+param newContainerImageName string?
+
+@description('Optional. The flag that indicates whether the existing image in the Container Registry should be overwritten.')
+param overwriteExistingImage bool = false
+
+@description('Optional. The cron expression that will be used to schedule the job.')
+@metadata({ default: '0 0 * * * // every day at midnight' })
+param cronExpression string = '0 0 * * *'
+
+@description('Optional. The CPU resources that will be allocated to the Container Apps Job.')
+param cpu string = '1'
+
+@description('Optional. The memory resources that will be allocated to the Container Apps Job.')
+param memory string = '2Gi'
+
+@description('Optional. The environment variables that will be added to the Container Apps Job.')
+@metadata({
+  example: '''[
+  {
+    name: 'ENV_VAR_NAME'
+    value: 'ENV_VAR_VALUE'
+  }
+  {
+    name: 'APPINSIGHTS_INSTRUMENTATIONKEY'
+    secretRef: 'applicationinsights-connection-string'
+  }
+]'''
+})
+param environmentVariables environmentVariablesType[]?
+
+import { secretType } from 'modules/deploy_services.bicep'
+@description('Optional. The secrets of the Container App. They will be added to Key Vault and configured as secrets in the Container App Job. The application insights connection string will be added automatically as `applicationinsightsconnectionstring`, if `appInsightsConnectionString` is set.')
+@metadata({
+  example: '''
+  [
+    {
+      name: 'mysecret'
+      identity: '/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myResourceGroup/providers/Microsoft.ManagedIdentity/userAssignedIdentities/myManagedIdentity'
+      keyVaultUrl: 'https://myvault${environment().suffixes.keyvaultDns}/secrets/mysecret'
+    }
+    {
+      name: 'mysecret'
+      identity: 'system'
+      keyVaultUrl: 'https://myvault${environment().suffixes.keyvaultDns}/secrets/mysecret'
+    }
+    {
+      // You can do this, but you shouldn't. Use a secret reference instead.
+      name: 'mysecret'
+      value: 'mysecretvalue'
+    }
+    {
+      name: 'connection-string'
+      value: listKeys('/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myResourceGroup/providers/Microsoft.Storage/storageAccounts/myStorageAccount', '2023-04-01').keys[0].value
+    }
+  ]
+  '''
+})
+#disable-next-line secure-secrets-in-params // @secure() is specified in UDT
+param secrets secretType[]?
+
+@description('Optional. Workload profiles for the managed environment. Leave empty to use a consumption based profile.')
+@metadata({
+  example: '''[
+    {
+      workloadProfileType: 'D4'
+      name: 'CAW01'
+      minimumCount: 0
+      maximumCount: 1
+    }
+  ]'''
+})
+param workloadProfiles array?
```

Comment:
Is this an array of objects? can this be typed?

Reply:
A UDT is best suited. Thanks for the hint.